zerver/tests/test_message_topics.py

```diff
@@ -333,3 +335,55 @@ def test_topic_delete_timeout(self) -> None:
             )
             result_dict = self.assert_json_success(result)
             self.assertFalse(result_dict["complete"])
+
+
+class EmptyTopicNameTest(ZulipTestCase):
+    def test_client_supports_empty_topic_name(self) -> None:
+        iago = self.example_user("iago")
+        hamlet = self.example_user("hamlet")
+        queue_data = dict(
+            all_public_streams=True,
+            apply_markdown=True,
+            client_type_name="website",
+            empty_topic_name=True,
+            event_types=["message"],
+            last_connection_time=time.time(),
+            queue_timeout=600,
+            realm_id=hamlet.realm.id,
+            user_profile_id=hamlet.id,
+        )
+        client = allocate_client_descriptor(queue_data)
+        self.assertTrue(client.event_queue.empty())
+
+        self.send_stream_message(iago, "Denmark", topic_name="")
+        events = client.event_queue.contents()
+        self.assertEqual(events[0]["message"]["subject"], "")
+
+        self.send_stream_message(iago, "Denmark", topic_name="general chat")
+        events = client.event_queue.contents()
+        self.assertEqual(events[1]["message"]["subject"], "")
+
+    def test_client_not_supports_empty_topic_name(self) -> None:
+        iago = self.example_user("iago")
+        hamlet = self.example_user("hamlet")
+        queue_data = dict(
+            all_public_streams=True,
+            apply_markdown=True,
+            client_type_name="zulip-mobile",
+            empty_topic_name=False,
+            event_types=["message"],
+            last_connection_time=time.time(),
+            queue_timeout=600,
+            realm_id=hamlet.realm.id,
+            user_profile_id=hamlet.id,
+        )
+        client = allocate_client_descriptor(queue_data)
+        self.assertTrue(client.event_queue.empty())
+
+        self.send_stream_message(iago, "Denmark", topic_name="")
+        events = client.event_queue.contents()
+        self.assertEqual(events[0]["message"]["subject"], "general chat")
+
+        self.send_stream_message(iago, "Denmark", topic_name="general chat")
+        events = client.event_queue.contents()
+        self.assertEqual(events[1]["message"]["subject"], "general chat")
```

Comment:
This test code could probably use `EMPTY_TOPIC_FALLBACK_NAME`

Reply:
Yeah, replaced them with `Message.EMPTY_TOPIC_FALLBACK_NAME`